pkg/runtime/serializer/codec_factory.go

```diff
@@ -44,24 +41,35 @@ type serializerType struct {
 	// be expected to pass into or a gvk to Decode, since no type information will be available on
 	// the object itself.
 	RawSerializer runtime.Serializer
-
 	// Specialize gives the type the opportunity to return a different serializer implementation if
 	// the content type contains alternate operations. Here it is used to implement "pretty" as an
 	// option to application/json, but could also be used to allow serializers to perform type
 	// defaulting or alter output.
 	Specialize func(map[string]string) (runtime.Serializer, bool)
+
+	AcceptStreamContentTypes []string
+	StreamContentType        string
+
+	Framer           runtime.Framer
+	StreamSerializer runtime.Serializer
+	StreamSpecialize func(map[string]string) (runtime.Serializer, bool)
 }
 
 func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory) []serializerType {
 	jsonSerializer := json.NewSerializer(mf, scheme, runtime.ObjectTyperToTyper(scheme), false)
 	jsonPrettySerializer := json.NewSerializer(mf, scheme, runtime.ObjectTyperToTyper(scheme), true)
 	serializers := []serializerType{
 		{
-			AcceptContentTypes: []string{"application/json"},
+			AcceptContentTypes: []string{"application/json", "application/json;stream=watch"},
```

Comment:
Yes, good point.

On Thu, Apr 21, 2016 at 6:07 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/runtime/serializer/codec_factory.go
> <https://github.com/kubernetes/kubernetes/pull/24539#discussion_r60556279>
> :
>
> >  }
> >
> >  func newSerializersForScheme(scheme *runtime.Scheme, mf json.MetaFactory) []serializerType {
> >  	jsonSerializer := json.NewSerializer(mf, scheme, runtime.ObjectTyperToTyper(scheme), false)
> >  	jsonPrettySerializer := json.NewSerializer(mf, scheme, runtime.ObjectTyperToTyper(scheme), true)
> >  	serializers := []serializerType{
> >  		{
> > -			AcceptContentTypes: []string{"application/json"},
> > +			AcceptContentTypes: []string{"application/json", "application/json;stream=watch"},
>
> Shouldn't this be the other way, i.e.:
> AcceptContentTypes accepts only "application/json"
>
> and AcceptStreamContentTypes accepts both?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24539/files/07c7ac06e50f992eea1af5088f67e9a3b1520b6a#r60556279>
>